frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,353 @@
+      return false;
+    }
+    return true;
+  };
+
+  nameValueArrayToObject = nvpArray => {
+    return _.reduce(nvpArray, (acc, nv) => _.assign(acc, { [nv[0]]: nv[1] }), {});
+  };
+  // Create PVC //
+  save = e => {
+    e.preventDefault();
+    if (!this.validateForm()) {
+      return;
+    }
+
+    const { kindObj, namespace } = this.props;
+    let resourceObj = { ...this.state.resourceObj };
+    let { pvcName, accessMode, requestSizeValue, requestSizeUnit, nameValuePairs } = this.state;
+
+    resourceObj.metadata.name = pvcName;
+    resourceObj.metadata.namespace = namespace;
+    resourceObj.spec.accessModes = [accessMode];
+    resourceObj.spec.resources.requests.storage = `${requestSizeValue}${requestSizeUnit}`;
+    //Code for label selectors
+    const valuePairs = this.nameValueArrayToObject(nameValuePairs);
+    if (!_.isEmpty(valuePairs)) {
+      _.set(resourceObj, 'spec.selector.matchLabels', valuePairs);
+    }
+    // Code for Storage Class selection
+    //   if (this.state.storageClass && this.state.storageClass.metadata.name !== "No Storage Class") {
+    //     //we can only have one storage class per claim
+    //     resourceObj.metadata.annotations["volume.beta.kubernetes.io/storage-class"] = this.state.storageClass.metadata.name;
+    //   }
+    this.setState({ resourceObj, inProgress: true });
+    k8sCreate(kindObj, resourceObj).then(
+      resource => {
+        this.setState({ inProgress: false });
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      },
+      err => this.setState({ error: err.message, inProgress: false })
+    );
+  };
+  render() {
+    const { defaultRequestSizeValue, defaultRequestSizeUnit, dropdownUnits, showKeyValue, nameValuePairs, inProgress, error } = this.state;
+    const title = 'Create Storage';
+
+    return (
+      <div className="co-m-pane__body">
+        <Helmet>
+          <title>{title}</title>
+        </Helmet>
+        <form className="co-m-pane__body-group " onSubmit={this.save}>
+          <h1 className="co-m-pane__heading">{title}</h1>
+
+          <div>
+            {/* <label className="control-label" htmlFor="volume-name">
+                  Storage Class
+                </label>
+                <StorageClassDropdown
+                  onChange={this.changeStorageClass}
+                  id="storageclass-dropdown"
+                  selectedKey={this.state.stringData.claimName}
+                />
+                <div className="form-input-padding" /> */}
+            <label className="control-label co-required" htmlFor="volume-name">
+              Name
+            </label>
+            <div className="form-group">
+              <input
+                className="form-control"
+                type="text"
+                onChange={this.handleChange}
+                placeholder="my-storage-claim"
+                aria-describedby="volume-name-help"
+                id="pvcName"
+                name="pvcName"
```

Comment:
name and id attribute should be separated by dash

Reply:
I use the name field to set the state variable using a generic handler function so I need the name to match.  I will add a dash to the ID.